### PR TITLE
chore(dev): automating binary creation for branches

### PR DESCRIPTION
**Pull request by allardy**

### CODE REVIEW

**Comment by samuelmasse** [comment]
What is this

**Comment by allardy** [comment]
I described this in my PR on the main repository here; 
https://github.com/botpress/botpress/pull/5434

Basically, when you push a commit to a branch, github actions will create a binary of messaging (exactly the same thing as when you release a new version of messaging, but scoped to a branch). 

Then, on the Botpress repository, you can add "devBranch" next to "version" in package.json, so if you checkout the branch in botpress, it will download the binary of the developer branch instead of the master version (so we can test it correctly without having to publish on master, then change the version)

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -10,54 +10,46 @@ jobs:
...
-        id: validate_semver
-        run: |
-          valid_latest_tag=$(node ./node_modules/semver/bin/semver.js ${{ steps.previous_tag.outputs.tag }})
-          valid_current_version=$(node ./node_modules/semver/bin/semver.js ${{ steps.extract_version.outputs.version }})
-          change_log=$(node -e "require('./scripts/changelog.js').build({ writeToFile: true }).then(cl => console.log(cl))")
 
-          # if not change log is truncated at first line
-          change_log="${change_log//'%'/'%25'}"
-          change_log="${change_log//$'\n'/'%0A'}"
-          change_log="${change_log//$'\r'/'%0D'}"
```
Comment (samuelmasse) [inline .github/workflows/release.yml:38]: Why was this changed?

File: .github/workflows/pre-release.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+name: Pre-Release
```
Comment (samuelmasse) [inline .github/workflows/pre-release.yml:1]: This config creates a new pre-release for every branch that is not master?

File: .github/workflows/branch_deleted.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+name: Branch deleted
```
Comment (samuelmasse) [inline .github/workflows/branch_deleted.yml:1]: Ok so here we delete the pre-release whenever a branch gets deleted.

**Review by samuelmasse — COMMENTED** [review]
Can you explain the changes a bit? Are the changes in release.yml just a refactor?

File: .github/workflows/pre-release.yml (hunk 2 above)
Comment (allardy) [reply .github/workflows/pre-release.yml:1]: Yes

File: .github/workflows/branch_deleted.yml (hunk 3 above)
Comment (allardy) [reply .github/workflows/branch_deleted.yml:1]: Yup, auto cleanup

File: .github/workflows/release.yml (hunk 1 above)
Comment (allardy) [reply .github/workflows/release.yml:38]: This code was copied here, in the studio and in messaging. I created a single action which extracts all the information we need and made the release process simpler. You can check the information here:

https://github.com/botpress/gh-actions/blob/master/get_release_details/src/index.ts

**Review by samuelmasse — APPROVED** [review]
